instrumentation/github.com/miekg/dns/splunkdns/test/client_test.go

```diff
@@ -0,0 +1,163 @@
+// Copyright Splunk Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package test
+
+import (
+	"context"
+	"errors"
+	"net"
+	"testing"
+	"time"
+
+	"github.com/miekg/dns"
+	"github.com/signalfx/splunk-otel-go/instrumentation/github.com/miekg/dns/splunkdns"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/sdk/trace"
+	"go.opentelemetry.io/otel/sdk/trace/tracetest"
+	traceapi "go.opentelemetry.io/otel/trace"
+)
+
+var defaultClientAttrs = []attribute.KeyValue{
+	attribute.String("default attribute", "client"),
+}
+
+func startServer(t *testing.T) *dns.Server {
+	// Let the system determine the port.
+	pc, err := net.ListenPacket("udp", "127.0.0.1:0")
+	require.NoError(t, err)
+
+	mux := dns.NewServeMux()
+	mux.HandleFunc(".", func(w dns.ResponseWriter, r *dns.Msg) {
+		m := new(dns.Msg)
+		m.SetReply(r)
+		_ = w.WriteMsg(m)
+	})
+
+	server := &dns.Server{
+		Addr:       pc.LocalAddr().String(),
+		Net:        pc.LocalAddr().Network(),
+		PacketConn: pc,
+		Handler:    mux,
+	}
+	errCh := make(chan error, 1)
+	go func() {
+		errCh <- server.ActivateAndServe()
+	}()
+	t.Cleanup(func() {
+		assert.NoError(t, <-errCh, "failed to start server")
```

Comment:
nit

```suggestion
		assert.NoError(t, <-errCh, "server.ActivateAndServe returned an error")
```

as there may be multiple reasons why it returns an error

moreover I hope this error message would make it easier to find where the error come from